Library/Homebrew/cmd/update.sh

```diff
@@ -336,23 +349,41 @@ EOS
         git fetch --force "${QUIET_ARGS[@]}" origin \
           "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH" 2>/dev/null
       else
-        git fetch --force "${QUIET_ARGS[@]}" origin \
-          "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH" || \
-            odie "Fetching $DIR failed!"
+        if ! git fetch --force "${QUIET_ARGS[@]}" origin \
+          "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH"
+        then
+          echo "Fetching $DIR failed!" >> "$HOMEBREW_UPDATE_FAILED_FILE"
```

Comment:
It's not appending e.g. a character at a time so it's actually fine. Unplug your internet and try yourself; definitely better than before (I wrote this on a ✈️ without internet 😉 )